src/dune_pkg/package_name.mli

```diff
@@ -1,8 +1,11 @@
 open! Stdune
```

Comment:
Since this is getting used in the rest of the source code, maybe it would make sense to rename this to `Opam_package_name`, because reading the diff a lot of the changes are replacing `Package.Name` with `Package_name` which seems like a rename but it actually has different semantics as it requires Opam-compatible names.

Reply:
Renamed to `Opam_compatible_package_name`.